src/components/services/patientService.ts

```diff
@@ -18,3 +18,24 @@ export const fetchPatients = async () => {
     } as User;
   });
 };
+
+export const fetchDoctors = async () => {
+  const doctorsCollectionRef = collection(db, "doctors");
+  const doctors = await getDocs(doctorsCollectionRef);
+
+  return doctors.docs.map((doc) => {
+    const docData = doc.data();
+    return {
+      id: doc.id,
+      name: docData.name || "",
+      speciality: docData.speciality || "",
+      email: docData.email,
+      degree: docData.degree || "",
+      address1: docData.address1 || "",
+      address2: docData.address2 || "",
+      experience: docData.experience || "",
+      fees: docData.fees || "",
+      about: docData.about || "",
+    };
+  });
```

Comment:
Nie jestem przekonany czy potrzebujemy tutaj w zasadzie duplikować te wartości, bo i klucze obiektu i wartości docData są tej samej nazwy. 

`return { id: doc.id, ...docData }`

Możesz też spróbować napisać funkcję pomocniczą, która przyjmuje dowolny obiekt i przechodzi po każdej własciwości i jeżeli jest nullem (bo to tutaj zabezpieczasz z tego co widzę || "") to wtedy podstawiasz jej pusty string. Będziesz mogła tej funkcji użwać w wielu miejscach + skróci się kod, bo zrobisz tylko `return fillEmptyValues({ id: doc.id, ...docData });`